plugins/calibre.koplugin/wireless.lua

```diff
@@ -644,9 +645,31 @@ end
 
 function CalibreWireless:sendToCalibre(arg)
     logger.dbg("GET_BOOK_FILE_SEGMENT", arg)
-    -- not implemented yet, we just send an invalid opcode to raise a control error in calibre.
-    -- If we don't do this calibre will wait *a lot* for the file(s)
-    self:sendJsonData('NOOP', {})
+    local inbox_dir = G_reader_settings:readSetting("inbox_dir")
+    local path = inbox_dir .. "/" .. arg.lpath
+
+    local file_size = lfs.attributes(path, "size")
+    if not file_size then
+        self:sendJsonData("NOOP", {})
+        return
+    end
+
+    local file = io.open(path, "rb")
+    if not file then
+        self:sendJsonData("NOOP", {})
+        return
+    end
+
+    self:sendJsonData("OK", { fileLength = file_size })
+
+    local chunk_size = 1024
```

Comment:
Would that be a touch small compared to 4K?

Reply:
Yeah, +1 for page-sized buffers (i.e., 4K ^^).